lib/msf/core/exploit/remote/smb/client/kerberos_authentication.rb

```diff
@@ -0,0 +1,162 @@
+# -*- coding: binary -*-
+
+#
+# This class implements an override for RubySMB's default authentication method to instead
+# use a kerberos authenticator
+#
+
+module Msf::Exploit::Remote::SMB::Client::KerberosAuthentication
+  def kerberos_authenticator=(kerberos_authenticator)
+    @kerberos_authenticator = kerberos_authenticator
+  end
```

Comment:
I don't have the full picture and I'm not sure if this will always be included in a `class`, but, can we use the `attr_writer` keyword instead?
```suggestion
  attr_writer :kerberos_authenticator
```